source/Ship.cpp

```diff
@@ -1913,6 +1913,16 @@ double Ship::JumpFuel() const
 
 
 
+double Ship::JumpFuelMissing() const
+{
+	// Used for smart refuelling: transfer only as much as really needed
+	// includes checking if fuel cap is high enough at all
+	if(!JumpsRemaining() && attributes.Get("fuel capacity") > JumpFuel())
+		return JumpFuel() - fuel;
```

Comment:
This function will report a compiler warning because it only returns a value in one control path. Also, I'm not sure what the purpose of the second check is.